test/dynamo/test_higher_order_ops.py

```diff
@@ -282,13 +282,183 @@ def false_fn2(x, y):
             name_set,
             {
                 "",
-                "cond_true_0",
-                "cond_false_0",
-                "cond_false_0.cond_false_0",
-                "cond_false_0.cond_true_0",
+                "cond_true_1",
+                "cond_false_1",
+                "cond_false_1.cond_false_0",
+                "cond_false_1.cond_true_0",
             },
         )
 
+    def test_cond_graph_break_in_one_branch(self):
+        backend = EagerAndRecordGraphs()
+        cnt = CompileCounterWithBackend(backend)
+
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.register_buffer("buffer", torch.ones(6, 4))
+
+            def forward(self, x):
+                def true_fn(x):
+                    self.buffer += 1
+                    return self.buffer.sum() + x.sum()
+
+                def false_fn(x):
+                    return (x - 1).sum()
+
+                return control_flow.cond(x.shape[0] > 4, true_fn, false_fn, [x])
+
+        mod_for_compile = torch.compile(Foo(), backend=cnt, dynamic=True)
+        mod_for_eager = Foo()
+
+        self.assertEqual(
+            mod_for_compile(torch.ones(6, 4)), mod_for_eager(torch.ones(6, 4))
+        )
+
```

Comment:
We should also assert that dynamo did graph break on this, by using `cnt` to count the number of frames. I'm not really sure what the number is, but the important thing is that it is > 1.